app/src/behaviors/behavior_sensor_rotate.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_sensor_rotate
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <drivers/sensor.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct behavior_sensor_rotate_config {
+    struct zmk_behavior_binding cw_binding;
+    struct zmk_behavior_binding ccw_binding;
+};
+
+struct behavior_sensor_rotate_data {
+    struct zmk_behavior_binding *triggered_binding;
+};
+
+static int on_sensor_binding_triggered(struct zmk_behavior_binding *binding,
+                                       const struct device *sensor, int64_t timestamp) {
+    const struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_sensor_rotate_config *cfg = dev->config;
+    struct behavior_sensor_rotate_data *data = dev->data;
+
+    struct sensor_value value;
+    int err;
+
+    err = sensor_channel_get(sensor, SENSOR_CHAN_ROTATION, &value);
+
+    if (err) {
+        LOG_WRN("Failed to get sensor rotation value: %d", err);
+        return err;
+    }
+
+    switch (value.val1) {
+    case 1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->cw_binding;
+        break;
+    case -1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->ccw_binding;
+        break;
+    default:
+        return -ENOTSUP;
+    }
+
+    LOG_DBG("Sensor binding: %s", log_strdup(binding->behavior_dev));
+
+    struct zmk_behavior_binding_event event = {.layer = zmk_keymap_highest_layer_active(),
+                                               .timestamp = timestamp};
+
+    behavior_keymap_binding_pressed(data->triggered_binding, event);
+    k_msleep(5);
+    behavior_keymap_binding_released(data->triggered_binding, event);
```

Comment:
You should use `zmk_behavior_queue_add()` so you can queue the press and release without blocking the main thread.

Reply:
Should the `k_msleep(5)` be kept in between the press and release with this method?

Reply:
No. The point is to allow scheduling the release to happen later without forcing the main thread to do nothing for 5 milliseconds. You can do

```c
zmk_behavior_queue_add(position, &binding, true, 0);
zmk_behavior_queue_add(position, &binding, false, 5);
```

I am not sure how to set the key position for a sensor though. The existing code here just uses 0 (by way of not setting the position when initializing `event`).

(We should probably also make a similar change to behavior_sensor_rotate_key_press.c in a separate PR.)

Reply:
Fully agreed on this, that API/infrastructure didn't exist when the original sensor behavior was written, which I imagine this was copied from.